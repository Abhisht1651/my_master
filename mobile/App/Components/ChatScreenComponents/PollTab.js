// @flow
import React from 'react'
import { View, Text, FlatList } from 'react-native'
import styles from '../Styles/ChatScreenComponentStyles/PollTabStyles'
import images from '../../Themes/Images'
import ShortPoll from '../ShortPoll'

type Props = {
  item: {
    question: string,
    votes: number,
    timeLeft: number,
    isPollActive: boolean,
    voterIcons: Array<string>,
    id: string,
    date: string,
    poll: {
      question: string,
      votes: number,
      timeLeft: number,
      isPollActive: boolean,
      voterIcons: Array<string>,
      id: string,
      date: string
    }
  }
}

const Data = [
  {
    id: '111',
    date: 'Today',
    poll: [
      {
        id: '1',
        question: 'Who are your favorites for the World Cup 2020?',
        votes: 54,
        timeLeft: 2,
        isPollActive: true,
        voterIcons: [images.voterIcon1, images.voterIcon2, images.voterIcon3]
      },
      {
        id: '2',
        question: 'Who are your favorites for the World Cup 2020?',
        votes: 54,
        timeLeft: 2,
        isPollActive: true,
        voterIcons: [images.voterIcon1, images.voterIcon2, images.voterIcon3]
      },
      {
        id: '3',
        question: 'Who are your favorites for the World Cup 2020?',
        votes: 54,
        timeLeft: 2,
        isPollActive: true,
        voterIcons: [images.voterIcon1, images.voterIcon2, images.voterIcon3]
      }
    ]
  },
  {
    id: '1111',
    date: 'Jan 15th 2020',
    poll: [
      {
        id: '11',
        question: 'Who are your favorites for the World Cup 2020?',
        votes: 54,
        timeLeft: 2,
        isPollActive: true,
        voterIcons: [images.voterIcon1, images.voterIcon2, images.voterIcon3]
      },
      {
        id: '12',
        question: 'Who are your favorites for the World Cup 2020?',
        votes: 54,
        timeLeft: 2,
        isPollActive: true,
        voterIcons: [images.voterIcon1, images.voterIcon2, images.voterIcon3]
      }
    ]
  }
]

const PollTab = () => {
  const renderpoll = ({ item }: Props) => {
    return (
      <View style={styles.storylinemargin}>
        <ShortPoll
          question={item.question}
          votes={item.votes}
          timeLeft={item.timeLeft}
          isPollActive={item.isPollActive}
          voterIcons={item.voterIcons}
        />
      </View>
    )
  }
  const renderitem = ({ item }: Props) => {
    return (
      <View>
        <View style={styles.flatlist1cont}>
          <Text style={styles.date}>{item.date}</Text>
          <FlatList
            data={item.poll}
            keyExtractor={(post) => post.id}
            renderItem={renderpoll}
            listKey={(post) => `${post.id}D`}
          />
        </View>
        <View style={styles.border} />
      </View>
    )
  }
  return (
    <View style={styles.container}>
      <View style={styles.margincontainer}>
        <FlatList
          data={Data}
          keyExtractor={(item) => item.id}
          renderItem={renderitem}
          listKey={(item) => `${item.id}D`}
        />
      </View>
    </View>
  )
}

export default PollTab
