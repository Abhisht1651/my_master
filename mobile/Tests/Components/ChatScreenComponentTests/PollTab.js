import React from 'react'
import { shallow } from 'enzyme'
import { FlatList } from 'react-native'
import PollTab from '../../../App/Components/ChatScreenComponents/PollTab'

describe('MessageScreen', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<PollTab />)
  })
  it('MediaTab component renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 1 flatlist', () => {
    expect(wrapper.find(FlatList)).toHaveLength(1)
  })
})
