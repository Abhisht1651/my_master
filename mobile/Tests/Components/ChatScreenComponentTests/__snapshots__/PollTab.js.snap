// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageScreen MediaTab component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F7F7F7",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginLeft": 50,
        "marginRight": 50,
      }
    }
  >
    <FlatList
      data={
        Array [
          Object {
            "date": "Today",
            "id": "111",
            "poll": Array [
              Object {
                "id": "1",
                "isPollActive": true,
                "question": "Who are your favorites for the World Cup 2020?",
                "timeLeft": 2,
                "voterIcons": Array [
                  Object {
                    "default": "",
                  },
                  Object {
                    "default": "",
                  },
                  Object {
                    "default": "",
                  },
                ],
                "votes": 54,
              },
              Object {
                "id": "2",
                "isPollActive": true,
                "question": "Who are your favorites for the World Cup 2020?",
                "timeLeft": 2,
                "voterIcons": Array [
                  Object {
                    "default": "",
                  },
                  Object {
                    "default": "",
                  },
                  Object {
                    "default": "",
                  },
                ],
                "votes": 54,
              },
              Object {
                "id": "3",
                "isPollActive": true,
                "question": "Who are your favorites for the World Cup 2020?",
                "timeLeft": 2,
                "voterIcons": Array [
                  Object {
                    "default": "",
                  },
                  Object {
                    "default": "",
                  },
                  Object {
                    "default": "",
                  },
                ],
                "votes": 54,
              },
            ],
          },
          Object {
            "date": "Jan 15th 2020",
            "id": "1111",
            "poll": Array [
              Object {
                "id": "11",
                "isPollActive": true,
                "question": "Who are your favorites for the World Cup 2020?",
                "timeLeft": 2,
                "voterIcons": Array [
                  Object {
                    "default": "",
                  },
                  Object {
                    "default": "",
                  },
                  Object {
                    "default": "",
                  },
                ],
                "votes": 54,
              },
              Object {
                "id": "12",
                "isPollActive": true,
                "question": "Who are your favorites for the World Cup 2020?",
                "timeLeft": 2,
                "voterIcons": Array [
                  Object {
                    "default": "",
                  },
                  Object {
                    "default": "",
                  },
                  Object {
                    "default": "",
                  },
                ],
                "votes": 54,
              },
            ],
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      listKey={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
</View>
`;
